### PR TITLE
Feat: Created timeline api

### PR DESCRIPTION
**Pull request by tsm1820**
# Notion Task

https://www.notion.so/Create-createTimelineEntry-api-2f80984417e24db1959f08e4089e2b48

# Changes Made In This PR

Added a createTimelineEntry function and useTimelineEntry function

# Also In This PR

Modified Timeline.ts in accordance with the latest database changes 

# Screenshots

No UI changes

### CODE REVIEW

File: src/api/timeline/createTimelineEntry.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import firebase from 'gatsby-plugin-firebase';
+import uuid from '../../utils/uuid';
+import { Entry } from '../../types/Timeline';
+import { User } from '../../types/User';
+
+const createTimelineEntry = async (args: {
+  user: User;
```
Comment (sheehui) [inline src/api/timeline/createTimelineEntry.ts:7]: Was wondering if it would be better to use `Timeline.id` here

Comment (millasml) [inline src/api/timeline/createTimelineEntry.ts:7]: since we only use the userId, can we just pass `userId` here?

Comment (millasml) [reply src/api/timeline/createTimelineEntry.ts:7]: `userId : User["uid"]`

File: src/types/Timeline.ts [hunk 2]
```diff
@@ -4,10 +4,11 @@ export type Timeline = {
 };
 
 export type Entry = {
-  isNew: boolean;
   resourceId: string;
-  date: string;
+  versionId: string;
```
Comment (millasml) [inline src/types/Timeline.ts:8]: make this an optional field

File: src/types/Timeline.ts [hunk 3]
```diff
@@ -4,10 +4,11 @@ export type Timeline = {
...
+  date: number;
   title: string;
   description: string;
   type: string;
+  updates: string[];
```
Comment (millasml) [inline src/types/Timeline.ts:13]: this should be an optional field too

**Review by millasml — CHANGES_REQUESTED** [review]
some small changes!

**Review by millasml — APPROVED** [review]

**Pull request merged**